jakarta.jdt/org.eclipse.lsp4jakarta.jdt.core/src/main/java/org/eclipse/lsp4jakarta/jdt/core/beanvalidation/BeanValidationDiagnosticsCollector.java

```diff
@@ -1,354 +1,281 @@
+                                DIAGNOSTIC_CODE_INVALID_TYPE, annotationName, DiagnosticSeverity.Error));
+                    }
+                } else if (matchedAnnotation.equals(PATTERN)) {
+                    if (!type.equals(getSignatureFormatOfType(STRING))
+                            && !type.equals(getSignatureFormatOfType(CHAR_SEQUENCE))) {
+                        diagnostics.add(createDiagnostic(element, declaringType.getCompilationUnit(),
+                                "The @" + annotationName
+                                        + " annotation can only be used on String and CharSequence type " + source,
+                                DIAGNOSTIC_CODE_INVALID_TYPE, annotationName, DiagnosticSeverity.Error));
+                    }
+                }
+
+                // These ones contains check on all collection types which requires resolving
+                // the String of the type somehow
+                // This will also require us to check if the field type was a custom collection
+                // subtype which means we
+                // have to resolve it and get the super interfaces and check to see if
+                // Collection, Map or Array was implemented
+                // for that custom type (which could as well be a user made subtype)
+
+//    			else if (annotation.getElementName().equals(NOT_EMPTY) || annotation.getElementName().equals(SIZE)) {
+//    				
+//    				System.out.println("--Field name: " + Signature.getTypeSignatureKind(fieldType));
+//    				System.out.println("--Field name: " + Signature.getParameterTypes(fieldType));			
+//    				if (	!fieldType.equals(getSignatureFormatOfType(CHAR_SEQUENCE)) &&
+//    						!fieldType.contains("List") &&
+//    						!fieldType.contains("Set") &&
+//    						!fieldType.contains("Collection") &&
+//    						!fieldType.contains("Array") &&
+//    						!fieldType.contains("Vector") &&
+//    						!fieldType.contains("Stack") &&
+//    						!fieldType.contains("Queue") &&
+//    						!fieldType.contains("Deque") &&
+//    						!fieldType.contains("Map")) {
+//    					
+//    					diagnostics.add(new Diagnostic(fieldAnnotationrange,
+//    							"This annotation can only be used on CharSequence, Collection, Array, "
+//    							+ "Map type fields."));	
+//    				}
+//    			}
+            }
+        }
+    }
+
+    /*
+     * Refer to Class signature documentation for the formating
+     * https://www.ibm.com/support/knowledgecenter/sl/SS5JSH_9.5.0/org.eclipse.jdt.
+     * doc.isv/reference/api/org/eclipse/jdt/core/Signature.html
+     */
+    private static String getSignatureFormatOfType(String type) {
+        return "Q" + type + ";";
+    }
+
+    private static String getDataTypeName(String type) {
+        int length = type.length();
+        if (length > 0 && type.charAt(0) == 'Q' && type.charAt(length - 1) == ';') {
+            return type.substring(1, length - 1);
+        }
+        return type;
+    }

```

Comment:
Any value in putting these helper methods in the `AbstractDiagnosticsCollector` class?

Reply:
Those are only used by this class so far, if they are reusable for others, I will move them to the parent.